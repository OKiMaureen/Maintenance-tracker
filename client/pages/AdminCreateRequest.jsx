import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import Header from '../components/Header';
import clearMessageAction from '../actions/clearMeassageAction';
import logoutAction from '../actions/logoutAction';
import Request from '../components/Request';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      const { clearMessage, requestDetail } = this.props;
      if (requestDetail.error) clearMessage();
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      const { history, logout } = this.props;
      logout();
      localStorage.clear();
      history.push('/');
      return true;
    }
    render() {
      const { requestDetail, userDetail } = this.props;
      const {
        title, description, equipment, errors, department,
      } = this.state;
      return (
        <div>
          <Header><Link to="/">Home</Link><Link to="/admincreaterequest">Create request</Link> <Link to="/adminrequests">All requests</Link><li><button className="header-btn" onClick={this.logout}>Logout</button></li></Header>
          <Request
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            title={title}
            description={description}
            equipment={equipment}
            requestDetail={requestDetail}
            userDetail={userDetail}
            errors={errors}
            department={department}
            titleText="Create Request"
            buttonText="Create Request"
          />
        </div>
      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
  clearMessage: clearMessageAction,
  logout: logoutAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

