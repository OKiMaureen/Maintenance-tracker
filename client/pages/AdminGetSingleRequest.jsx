import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import i from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import adminGetSingleRequestAction, { updateReq, disapproveRequest, resolveRequest } from '../actions/adminSingleRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AdminSingleRequest extends Component {
  componentDidMount() {
    const { adminSingleRequest, match } = this.props;
    adminSingleRequest(match.params.id);
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  updateRequest = (e) => {
    const status = e.target.id;
    const { update, oneRequest } = this.props;
    update(oneRequest.id, status);
    return true;
  }

  render() {
    const {
      oneRequest,
    } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/adminrequests">All requests</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <section className="allrequests">
          {oneRequest ?
            <div className="detailsrequest-card" key={i.generate()}>
              <div className="status">
                {oneRequest.requeststatus === 'approved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="approve" className="disabled" disabled onClick={this.updateRequest}>Approve</button> :
                  <button className="green" id="approve" onClick={this.updateRequest}>Approve</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="disapprove" className="disabled" disabled onClick={this.updateRequest}>Disapprove</button> :
                  <button className="red" id="disapprove" onClick={this.updateRequest}>Disapprove</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' || oneRequest.requeststatus === 'pending' ?
                  <button id="resolved" className="disabled" disabled onClick={this.updateRequest}>Resolve</button> :
                  <button className="green" id="resolve" onClick={this.updateRequest}>Resolve</button>}
              </div>
              <p>
                <label>Title:</label>{oneRequest.title}
              </p>
              <p>
                <label>Department:</label>{oneRequest.department}
              </p>
              <p>
                <label>Equipment: </label>{oneRequest.equipment}
              </p>
              <p>
                <label>Description: </label>{oneRequest.description}
              </p>
              <div className="container">
                <div className="status fixed">
                  {oneRequest.requeststatus === 'pending' ? <label className="yellow">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'approved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'disapproved' ? <label className="red">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'resolved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                </div>
              </div>
            </div> : '' }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  oneRequest: state.adminSingleRequest.request,
  userDetail: state.authUser,
  logout: logoutAction,

});
const mapDispatchToProps = dispatch => bindActionCreators({
  adminSingleRequest: adminGetSingleRequestAction,
  update: updateReq,
  disapprove: disapproveRequest,
  resolve: resolveRequest,

}, dispatch);

AdminSingleRequest.propTypes = {
  oneRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  adminSingleRequest: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AdminSingleRequest);
