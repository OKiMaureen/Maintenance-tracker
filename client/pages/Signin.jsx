import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import signInAction from '../actions/signInAction';
import loader from '../assets/images/spinloader.gif';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class Signin extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      email: '',
      password: '',
      errors: {},
    }

    validate = () => {
      const validations = new Validator(
        {
          email: this.state.email,
          password: this.state.password,
        },
        {
          email: 'required|email|string',
          password: 'required|min:8|max:30|string',

        },
        {
          'min.password': 'The :attribute must not be less than 9 characters.',
          'max.password': 'The :attribute must not be greater than 30 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.signInAction(this.state, this.props.history);
      }
      return true;
    }

    render() {
      return (
        <div>
          <Header><Link to="/">Home</Link><Link to="/" /></Header>
          {/* Signup Form  */}
          <section className="form">
            <article className="card">
              <form id="signupForm">
                <h3>Sign In</h3>
                {/* <LoadingBubble /> */}
                <br />
                {/* <img src={loader} alt="loader" /> */}
                {this.props.userDetail.checkStatus.isLoading ? (<span className="loader"><img src={loader} alt="loader" /></span>) : '' }
                {
                  this.props.userDetail.error ? <span className="validation-error">{this.props.userDetail.error}</span> : ''
                }
                <input type="text" id="email" name="email" placeholder="Email" onChange={this.handleChange} required />
                {
                  this.state.errors.email ? this.state.errors.email.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }

                <br />
                <input type="password" id="password" name="password" placeholder="Password" onChange={this.handleChange} required />
                {
                  this.state.errors.password ? this.state.errors.password.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
                <br />
                <button type="submit" className="button" id="signupBtn" onClick={this.handleSubmit} >Sign In</button>
                <p>Donot have an account?
                  <Link to="/signup">Sign Up</Link>
                </p>
              </form>
            </article>
          </section>
          {/* End signup form */}
          <Footer />
        </div>

      );
    }
}

const mapStateToProps = state => ({
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  signInAction,
}, dispatch);

Signin.propTypes = {
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  signInAction: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(Signin);

