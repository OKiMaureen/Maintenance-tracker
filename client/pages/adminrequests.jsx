import React from 'react';

const AdminRequests = () => ((
    <div>
        <section className="approverequests">
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
            
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
         </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
           </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
      </section>
    </div>
));
export default AdminRequests;