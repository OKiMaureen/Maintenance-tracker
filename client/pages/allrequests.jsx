import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import id from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import allRequestAction from '../actions/allRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AllRequests extends Component {
  componentDidMount() {
    const { allRequest } = this.props;
    allRequest();
  }
  logout = () => {
    const { history, logout } = this.props;
    if (logout()) {
      localStorage.clear();
      history.push('/');
    }
    return true;
  }
  render() {
    const { requests = [] } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/allrequests" className="current">My requests</Link>
          <button className="header-btn" onClick={this.logout}>Logout</button>
        </Header>

        <section className="allrequests">
          {requests && requests.length !== 0 ? requests.map(request =>

          (
            <div className="requests-card" key={id.generate}>
              <p>
                <label>Title:</label>{request.title}
              </p>
              <p>
                <label>Equipment: </label>{request.equipment}
              </p>
              <div className="container">
                <div className="status fixed">
                  {request.requeststatus === 'pending' ? <label className="yellow">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'approved' ? <label className="green">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'disapproved' ? <label className="red">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'resolved' ? <label className="green">{request.requeststatus}</label> : ''}
                </div>
                <div className="status flex-item ">
                  <p>
                    { <Link to={`/singlerequest/${request.id}`}> More</Link>}

                  </p>
                </div>
              </div>
            </div>)) :
          <div className="requests-card no-request" >
            <p className="message-centered">
              You do not have any requests yet!!! Go ahead and create a request.
            </p>
          </div>
          }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  requests: state.requests.requests,
  error: state.requests.error,
  userDetail: state.authUser,

});


const mapDispatchToProps = dispatch => bindActionCreators({
  allRequest: allRequestAction,
  logout: logoutAction,
}, dispatch);

AllRequests.propTypes = {
  requests: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  allRequest: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AllRequests);
