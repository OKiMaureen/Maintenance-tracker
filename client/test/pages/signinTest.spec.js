import React from 'react';
import { expect as e } from 'chai';
import { shallow } from 'enzyme';
import { Signin } from '../../pages/Signin';


describe('<SignIn />', () => {
  let wrapper;
  const props = {
    signInAction: () => {},
    userDetail: {
      authUser: {
        id: 1,
        name: 'maureen',
        email: 'ccureen.eloho@gmail.com',
        role: 'user',
      },
      checkStatus: {
        isLoading: false,
        success: false,
        error: false,
      },
    },
    history: {},
    error: '',
  };
  it('Should return number of input field on Signin page', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.find('input')).to.have.length(2);
  });
  it('stores a snapshot of the component', () => {
    wrapper = shallow(<Signin {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should return number of button on Signin page', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.find('button')).to.have.length(1);
  });
  it('Should check if the wrapper contains instance of Signin page', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.instance()).to.be.instanceof(Signin);
  });

  it('the validator function should return true when validation is successfull', () => {
    wrapper = shallow(<Signin {...props} />);
    wrapper.setState({
      email: 'maureen@mail.ru',
      password: 'password',
    });
    const { validate } = wrapper.instance();

    const result = validate();

    expect(result).toBe(true);
  });
  it('the handlechange function should clear the errors  when valid input is written', () => {
    wrapper = shallow(<Signin {...props} />);
    wrapper.setState({
      email: 'maureen@mail.com',
      password: 'password',
      errors: {
        password: '',
      },
    });
    const { handleChange } = wrapper.instance();
    handleChange({
      target: {
        value: 'password',
        name: 'password',
      },
    });

    expect(wrapper.state().errors.password).toBe('');
  });
  it('Should check if email is supplied on successful Signin', () => {
    wrapper = shallow(<Signin {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, email: 'maureen@gmail.com' });
    e(wrapper.state().email).to.be.equal('maureen@gmail.com');
  });
  it('Should check if password is supplied on successful Signin', () => {
    wrapper = shallow(<Signin {...props} />);
    const initialState = wrapper.state();
    wrapper.setState({ ...initialState, password: 'maureen' });
    e(wrapper.state().password).to.be.equal('maureen');
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'email' } })).to.be.equal(true);
  });
  it('Should handle and store user details', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.instance().handleChange({ target: { value: 'password' } })).to.be.equal(true);
  });

  it('Should handle user details validation', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.instance().validate()).to.be.equal(false);
  });

  it('Should handle signing in a user', () => {
    wrapper = shallow(<Signin {...props} />);
    e(wrapper.instance().handleSubmit({ preventDefault: () => {} })).to.be.equal(true);
  });
  it('Should handle loading a loader', () => {
    wrapper = shallow(<Signin
      {...props}
      userDetail={{
      checkStatus: {
        isLoading: true,
      },
    }}
    />);
    expect(wrapper.find('.loader').length).toBe(1);
  });
  it('Should handle loading a not loading on success or error', () => {
    wrapper = shallow(<Signin
      {...props}
      userDetail={{
      checkStatus: {
        isLoading: false,
      },
    }}
    />);
    expect(wrapper.find('.loader').length).toBe(0);
  });
  it('Should handle loading the error message', () => {
    wrapper = shallow(<Signin
      {...props}
      userDetail={{
        checkStatus: {
          isLoading: false,
      },

      error: 'Email already exists',
    }}
    />);
    expect(wrapper.find('.validation-error').length).toBe(1);
  });
  it('Should handle  not loading the error message on success or loading', () => {
    wrapper = shallow(<Signin
      {...props}
      userDetail={{
        checkStatus: {
          isLoading: false,
      },

      error: '',
    }}
    />);
    expect(wrapper.find('.validation-error').length).toBe(0);
  });
  it('Should handle submit method', () => {
    const actual = wrapper.instance().handleSubmit({ preventDefault: () => {} });
    expect(actual).toBeDefined();
  });
  it('Should handle submit', () => {
    wrapper = shallow(<Signin {...props} />);
    const event = {
      preventDefault: () => {},
    };
    expect(wrapper.instance().handleSubmit(event)).toBe(true);
  });
});
