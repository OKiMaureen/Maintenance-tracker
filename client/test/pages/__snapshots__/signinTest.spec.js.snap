// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> stores a snapshot of the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Signin
    error=""
    history={Object {}}
    signInAction={[Function]}
    userDetail={
      Object {
        "authUser": Object {
          "email": "ccureen.eloho@gmail.com",
          "id": 1,
          "name": "maureen",
          "role": "user",
        },
        "checkStatus": Object {
          "error": false,
          "isLoading": false,
          "success": false,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="form"
      >
        <article
          className="card"
        >
          <form
            id="signupForm"
          >
            <h3>
              Sign In
            </h3>
            <br />
            
            
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="text"
            />
            
            <br />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            
            <br />
            <button
              className="button"
              id="signupBtn"
              onClick={[Function]}
              type="submit"
            >
              Sign In
            </button>
            <p>
              Donot have an account?
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </p>
          </form>
        </article>
      </section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <article
          className="card"
        >
          <form
            id="signupForm"
          >
            <h3>
              Sign In
            </h3>
            <br />
            
            
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="text"
            />
            
            <br />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            
            <br />
            <button
              className="button"
              id="signupBtn"
              onClick={[Function]}
              type="submit"
            >
              Sign In
            </button>
            <p>
              Donot have an account?
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </p>
          </form>
        </article>,
        "className": "form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            id="signupForm"
          >
            <h3>
              Sign In
            </h3>
            <br />
            
            
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="text"
            />
            
            <br />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            
            <br />
            <button
              className="button"
              id="signupBtn"
              onClick={[Function]}
              type="submit"
            >
              Sign In
            </button>
            <p>
              Donot have an account?
              <Link
                replace={false}
                to="/signup"
              >
                Sign Up
              </Link>
            </p>
          </form>,
          "className": "card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Sign In
              </h3>,
              <br />,
              "",
              "",
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />,
              "",
              <br />,
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />,
              "",
              <br />,
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign In
              </button>,
              <p>
                Donot have an account?
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>,
            ],
            "id": "signupForm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "",
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "required": true,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "required": true,
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "className": "button",
                "id": "signupBtn",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Donot have an account?",
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Donot have an account?",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
              ],
              "type": "p",
            },
          ],
          "type": "form",
        },
        "type": "article",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="form"
        >
          <article
            className="card"
          >
            <form
              id="signupForm"
            >
              <h3>
                Sign In
              </h3>
              <br />
              
              
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign In
              </button>
              <p>
                Donot have an account?
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>
            </form>
          </article>
        </section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <article
            className="card"
          >
            <form
              id="signupForm"
            >
              <h3>
                Sign In
              </h3>
              <br />
              
              
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign In
              </button>
              <p>
                Donot have an account?
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>
            </form>
          </article>,
          "className": "form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              id="signupForm"
            >
              <h3>
                Sign In
              </h3>
              <br />
              
              
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="text"
              />
              
              <br />
              <input
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
              />
              
              <br />
              <button
                className="button"
                id="signupBtn"
                onClick={[Function]}
                type="submit"
              >
                Sign In
              </button>
              <p>
                Donot have an account?
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </p>
            </form>,
            "className": "card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Sign In
                </h3>,
                <br />,
                "",
                "",
                <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="text"
                />,
                "",
                <br />,
                <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />,
                "",
                <br />,
                <button
                  className="button"
                  id="signupBtn"
                  onClick={[Function]}
                  type="submit"
                >
                  Sign In
                </button>,
                <p>
                  Donot have an account?
                  <Link
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </p>,
              ],
              "id": "signupForm",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign In",
                },
                "ref": null,
                "rendered": "Sign In",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "",
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "required": true,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "required": true,
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign In",
                  "className": "button",
                  "id": "signupBtn",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Sign In",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Donot have an account?",
                    <Link
                      replace={false}
                      to="/signup"
                    >
                      Sign Up
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Donot have an account?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign Up",
                      "replace": false,
                      "to": "/signup",
                    },
                    "ref": null,
                    "rendered": "Sign Up",
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
            ],
            "type": "form",
          },
          "type": "article",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
